src/main/java/org/jenkinsci/plugins/durabletask/executors/OnceRetentionStrategy.java

```diff
@@ -56,6 +56,18 @@ public OnceRetentionStrategy(int idleMinutes) {
         super(idleMinutes);
     }
 
+    @Override
+    public long check(final AbstractCloudComputer c) {
+        // When the slave is idle we should disable accepting tasks and check to see if it is already trying to
+        // terminate. If it's not already trying to terminate then lets terminate manually.
+        if (c.isIdle()) {
+            done(c);
```

Comment:
This is bypassing the `idleMinutes` condition previously done in the superclass, so whatever its other benefits, it cannot be correct. I suspect what you meant this method to be was something like

```java
        if (c.isIdle() && !disabled) {
            final long idleMilliseconds = System.currentTimeMillis() - c.getIdleStartMilliseconds();
            if (idleMilliseconds > MINUTES.toMillis(idleMinutes)) {
                LOGGER.log(LEVEL.FINE, "Disconnecting {0}", c.getName());
                done(c);
            }
        }
        return 1;
```

with a workaround for the fact that `CloudRetentionStrategy.idleMinutes` is `private`, either by shadowing the field here, or just extending `RetentionStrategy<AbstractCloudComputer>` directly (and inlining `super.start`).